src/components/views/rooms/AppsDrawer.js

```diff
@@ -162,6 +164,18 @@ module.exports = React.createClass({
             e.preventDefault();
         }
 
+        // Display a warning dialog if the max number of widgets have already been added to the room
```

Comment:
This feels like one too many clicks for the user to make. Have we considered displaying something else instead of "Add a widget" ? Perhaps "Maximum number of widgets reached"?